bfsdk/cmake/CMakeGlobal_Project.txt

```diff
@@ -16,147 +16,33 @@
-    else()
-        set(CMAKE_BUILD_TYPE "Release")
-    endif()
+if(ENABLE_UNITTESTING)
+    include(CTest)
+    enable_testing(true)
 endif()
 
 if(ENABLE_TIDY AND NOT WIN32)
     set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 endif()
 
-if(NOT DISABLE_CTEST AND ENABLE_UNITTESTING)
-    enable_testing(true)
-endif()
-
-# ------------------------------------------------------------------------------
-# Library Types
-# ------------------------------------------------------------------------------
-
-option(BUILD_SHARED_LIBS "" OFF)
-option(BUILD_STATIC_LIBS "" ON)
-
-if(ENABLE_UNITTESTING)
-    if(BUILD_SHARED_LIBS)
-        set(BAREFLANK_HOST_SHARED_LIBS ON)
-        set(BAREFLANK_HOST_STATIC_LIBS OFF)
-        set(BAREFLANK_TEST_SHARED_LIBS OFF)
-        set(BAREFLANK_TEST_STATIC_LIBS ON)
-    else()
-        set(BAREFLANK_HOST_SHARED_LIBS OFF)
-        set(BAREFLANK_HOST_STATIC_LIBS ON)
-        set(BAREFLANK_TEST_SHARED_LIBS OFF)
-        set(BAREFLANK_TEST_STATIC_LIBS ON)
-    endif()
-else()
-    if(BUILD_SHARED_LIBS)
-        set(BAREFLANK_HOST_SHARED_LIBS ON)
-        set(BAREFLANK_HOST_STATIC_LIBS OFF)
-        set(BAREFLANK_TEST_SHARED_LIBS OFF)
-        set(BAREFLANK_TEST_STATIC_LIBS OFF)
-    else()
-        set(BAREFLANK_HOST_SHARED_LIBS OFF)
-        set(BAREFLANK_HOST_STATIC_LIBS ON)
-        set(BAREFLANK_TEST_SHARED_LIBS OFF)
-        set(BAREFLANK_TEST_STATIC_LIBS OFF)
-    endif()
-endif()
-
-message(STATUS "Build Shared Libs: ${BUILD_SHARED_LIBS}")
-message(STATUS "Build Static Libs: ${BUILD_STATIC_LIBS}")
-
-# ------------------------------------------------------------------------------
-# Detect 32bit
-# ------------------------------------------------------------------------------
-
-if(NOT CMAKE_CXX_SIZEOF_DATA_PTR STREQUAL "8")
-    message(FATAL_ERROR "Only 64bit builds are supported")
-endif()
-
-# ------------------------------------------------------------------------------
-# Install Symlink
-# ------------------------------------------------------------------------------
-
-macro(install_symlink filepath sympath)
-    if(WIN32)
-        install(CODE "execute_process(COMMAND mklink ${sympath} ${filepath})")
-        install(CODE "message(STATUS \"Created symlink: ${sympath} -> ${filepath}\")")
-    else()
-        install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink ${filepath} ${sympath})")
-        install(CODE "message(STATUS \"Created symlink: ${sympath} -> ${filepath}\")")
-    endif()
-endmacro(install_symlink)
-
-# ------------------------------------------------------------------------------
-# Prefix
-# ------------------------------------------------------------------------------
-
-message(STATUS "Install path: ${CMAKE_INSTALL_PREFIX}")
-
 # ------------------------------------------------------------------------------
 # OS
 # ------------------------------------------------------------------------------
 
+# TODO: Use CMAKE_SYSTEM_NAME to detect your host OS, and allow for build
```

Comment:
That will not work on Cygwin as the toolchain is what defines this, and we need to know if we are on Windows / Cygwin, even if we get a toolchain that defines "Linux" which is our target for the cross compiled components.

Reply:
Looks like I misread the docs on this one, good catch. I think that CMAKE_HOST_SYSTEM_NAME is the correct variable I was looking for.

Reply:
Huh, if that works I would be very happy